packages/docusaurus-mdx-loader/src/remark/relativePath/index.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const visit = require('unist-util-visit');
+
+const plugin = () => {
+  const transformer = (root) => {
+    visit(root, 'image', (node) => {
+      if (node.url.startsWith('.')) {
```

Comment:
wonder if we couldn't support other patterns that do not start with .

Reply:
I was thinking of appending static dir in case of an absolute path

Reply:
I think, this is `![](./assets/docusaurus-asset-example-banner.png)` more clear than `![](assets/docusaurus-asset-example-banner.png)`

Reply:
Yes, I also prefer `./`, but I'd say we should support both anyway, to avoid being confusing for users. Not using the `./` prefix is also valid pathname resolution => https://github.com/mjackson/resolve-pathname

I think we should still convert `/img/myImage.png` to `require('/staticDirPath/myImage.png')` by default, even if the file does not exist (intentional to have a build compilation error). Filtering transforms using fs.exists(path) would not give compilation errors.

But this is a risky move, and would also enforce using a webpack loader for all images (and webpack would apply a hash on this image in the build folder, which can be annoying)

So I think we should offer an escape hatch to the users, to remove that risk, and give them the ability to use the existing/old image path behavior. For example, we could support an escape syntax like `![](static:/img/myImage.png)`, in such case, we would not use any webpack loader (so, no compile-time security), and would just use the path provided by the user as-is.

For images of the `/static` folder, I think we should encourage the users to co-locate more the images to their markdown usage, so this folder would only be for rare cases where an image is displayed in multiple docs, or when the user don't want to use the webpack loader for any kind of reason (there might be usecases where the user copy a bunch of images to the output folder only after builds etc...).

Reply:
```js
if (!fs.existsSync(path.join(options.staticDir, node.url))) {
          throw new Error(`File ${node.url} not found in ${options.staticDir}`);
        }

```
We can throw error if file doesn't exists

Reply:
require('assets/docusaurus-asset-example-banner.png') fails

Reply:
Yes, should probably throw a custom error instead of having the default webpack error: gives more opportunity to help the user.

> require('assets/docusaurus-asset-example-banner.png') fails

That's right, but I think `![](assets/docusaurus-asset-example-banner.png)` is valid markdown, so in such case you should add the `./` prefix to the require call so that it works with webpack too :)

Reply:
![image](https://user-images.githubusercontent.com/32068075/87952649-b27c1380-cac7-11ea-8a68-48a10d9e0c33.png)
Are these errors fine?

Reply:
yes that looks nice